### DISCUSSION TITLE
Built-In ADC for Low Bandwidth Telephony Over LAN

### DISCUSSION THREAD

**Discussion by CouchDweller** [pschatzmann/arduino-audio-tools#236] [Q&A] [answered]
Howdy,

I would like to use the ESP32 built-in ADC to acquire low-bandwith (~4kHz, ~8K samps/sec) analog telephony audio and send audio over my LAN.

I have already been successful sending the test tone per...
arduino-audio-tools/examples/examples-communication/communication-ip-send/communication-ip-send.ino

Now trying to figure out how to substitute...
GeneratedSoundStream<int16_t> sound( sineWave);  // Stream generated from sine wave
...with commands to use the ADC (also specifying sample rate, amplitude scaling, and sample bitness), but can't quite figure out how.

Any advice would be super duper appreciated.

Cheers,

CD

**Comment by pschatzmann** [comment, marked as answer]
You would use a [AnalogAudioStream](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_analog_audio_stream.html) as input and to manage the volume you can use the logic described in [VolumeControl](https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control)

You might also want to consider to use a [codec](https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio)
Please note that the AnalogAudioStream is providing 2 channels and [you might want to reduce it to one](https://github.com/pschatzmann/arduino-audio-tools/wiki/Converting-the-Number-of-Input-and-Output-Channels)
